src/pages/core/external-plugins/app-data.md

```diff
@@ -248,3 +349,178 @@ let write_to_app_data_plugin_ix = WriteExternalPluginAdapterDataV1CpiBuilder::ne
+
+```rust
+
+let plugin_authority = ctx.accounts.authority.key();
+
+let asset = BaseAssetV1::from_bytes(&data).unwrap();
+
+// fetches the AppData plugin based on the Authority of the plugin
+let plugin_key = ExternalPluginAdapterKey::AppData(PluginAuthority::Address {
+    address: plugin_authority });
+
+let app_data_plugin = fetch_external_plugin_adapter::<BaseAssetV1, AppData>(
+        &account_info,
+        Some(&base_asset),
+        &plugin_key,
+    )
+    .unwrap();
+
+let (data_offset, data_length) =
+        fetch_external_plugin_adapter_data_info(&account_info, Some(&asset), &plugin_key)
+            .unwrap();
+
+// grab app_data data from account_info
+let data = account_info.data.borrow()[data_offset..data_offset + data_length].to_vec();
+
+```
+
+{% /dialect %}
+
+{% /dialect-switcher %}
+
+### Deserialization
+
+Now that you have the data you'll need to deserialize the data depending on the schema you chose to write the data with to the AppData plugins.
+
+#### Deserialze JSON Schema
+
+For the **JSON** schema you will need to use the `serde` and `serde_json` crates.
+
+{% dialect-switcher title="Deserializing JSON" %}
+{% dialect title="JavaScript" id="js" %}
+
+```ts
+// decode the data into a string
+const jsonString = new TextDecoder().decode(data)
+//pasrse the string into a useable Javascript object
+const object = JSON.parse(string)
+```
+
+{% /dialect %}
+
+{% dialect title="Rust" id="rust" %}
+
+```rust
+#[derive(Debug, Serialize, Deserialize)]
+pub struct MyData {
+    pub timestamp: u64,
+    pub message: String,
+}
+
+
+// deserialization from json to struct
+// this uses `serde` and `serde_json` crates
+let my_data: MyData = serde_json::from_slice(&data).unwrap();
+println!("{:?}", my_data);
+```
+
+{% /dialect %}
+
+{% /dialect-switcher %}
+
+#### Deserialze MsgPack Schema
+
+For the **MsgPack** schema you will need to use the `serde` and `rmp_serde` crates.
+
+{% dialect-switcher title="Deserializing MsgPack" %}
+{% dialect title="JavaScript" id="js" %}
+
+```ts
+// decode the data into a Javascript object using `msgpack-lite`
+const object = msgpack.decode(bytes)
+```
+
+{% /dialect %}
+
+{% dialect title="Rust" id="rust" %}
+
+```rust
+#[derive(Debug, Serialize, Deserialize)]
+pub struct MyData {
+    pub timestamp: u64,
+    pub message: String,
+}
+
+// deserialization from msgpack to struct
+// this uses `serde` and `rmp_serde` crates
+let my_data: MyData = rmp_serde::decode::from_slice(&data).unwrap();
+println!("{:?}", my_data);
+```
+
+{% /dialect %}
+
+{% /dialect-switcher %}
+
+#### Deserialze Binary Schema
+
+Because the **Binary** schema is arbitory data then deserialization will be dependent on the serialization you used.
+
+{% dialect-switcher title="Deserializing MsgPack" %}
+{% dialect title="JavaScript" id="js" %}
+```js
+As the binary data is arbitory you will need to include your own deserializers to
+parse the data into a usable format your app/website will understand.
```

Comment:
Fix in future commit for switcher Title.

This was for JS, we don't have bincode in JS that I know of. How would you want to show Javascript deserialization and serialization of bytes? For now I just left it open ended as people wanting using binary serialization and deserialization I would assume would have their methods already chosen.

In the serialization example I just had a `Uinit8Array([1,2,3,4,5])` that was stored to data.

Reply:
Would it be beneficial to show bincode in JS the way it is shown in Rust?  Or if you are sticking with the array example you could show just like an assert or log or something just showing accessing the Uint8Array after its gotten?

This could also be handled based on what the JS deserialization examples look like based on the other comment about how `fetchAsset` already deserializes the data.